drivers/i2c/CMakeLists.txt

```diff
@@ -12,7 +12,13 @@ zephyr_library_sources_ifdef(CONFIG_I2C_XEC		i2c_mchp_xec.c)
 zephyr_library_sources_ifdef(CONFIG_I2C_MCUX		i2c_mcux.c)
 zephyr_library_sources_ifdef(CONFIG_I2C_MCUX_LPI2C	i2c_mcux_lpi2c.c)
 zephyr_library_sources_ifdef(CONFIG_NRFX_TWI		i2c_nrfx_twi.c)
+
+if(CONFIG_I2C_USE_MNGR)
+zephyr_library_sources_ifdef(CONFIG_NRFX_TWIM		i2c_ll_nrfx_twim.c)
+else()
 zephyr_library_sources_ifdef(CONFIG_NRFX_TWIM		i2c_nrfx_twim.c)
+endif()
```

Comment:
Are you sure you want to do that? Unconditionally enabling i2c_mngr without letting other part of the code using the regular API is probably not a good idea. You cannot be sure some part of the system is still using the regular API vs this one. 

Plus, wasn't it so your API was supposed to use the regular API without requiring any driver changes?